### ISSUE TITLE
Assess number of zookeeper watches being set by different processes types.

### ISSUE THREAD

**Issue by keith-turner** [apache/accumulo#5134]
Accumulo code is structured to minimize the number zookeeper connections and number of zookeeper watches.  With the large number of changes made in 4.0 it is possible that these goals are not being met as well as they used to be.  Developing a way to measure the number of connections and watches per process would help determine if what is happening with this goal.   Would be nice to be able to see this information per server and per client.  There may not be any accumulo changes needed for this, may be able to do this w/ existing zookeeper mechanisms and some scripts.

**Comment by keith-turner** [comment]
Accumulo will select random servers for some client operations.  If there are lot of servers and client is making lots of API calls that select a random servers then what happens w/ ZK watches in that client over time?  This is an example of a scenario where it would be nice to have data for clients in addition to servers.

**Comment by keith-turner** [comment]
This zookeeper client has class for inspecting a clients watches.  Maybe accumulo processes could be made to call that and dump their watches.   Could also setup metrics in each process to report the number of watches it has.

https://zookeeper.apache.org/doc/current/apidocs/zookeeper-server/org/apache/zookeeper/server/watch/WatchManager.html

**Comment by keith-turner** [comment]
Discovered WatchManager is not public API for zookeeper.  So may not be able to get a count of watches for a zookeeper object.

**Comment by keith-turner** [comment]
This comment proposed a potential way for zoocache to deal with entries read from zoocache only once.   This may or may not actually be a problem.  Need to collect data on how much this happens in a live Accumulo system and go from there.  That data collection could possibly be done as part of this issue.

https://github.com/apache/accumulo/pull/5143#discussion_r1873997910

**Comment by dlmarion** [comment]
@keith-turner  - Do you think we can close this now that #5256 is merged?
